src/dtgtk/thumbtable.c

```diff
@@ -2226,12 +2231,17 @@ dt_thumbtable_t *dt_thumbtable_new()
                    G_CALLBACK(_event_leave_notify), table);
   g_signal_connect(G_OBJECT(table->widget), "enter-notify-event",
                    G_CALLBACK(_event_enter_notify), table);
-  g_signal_connect(G_OBJECT(table->widget), "button-press-event",
-                   G_CALLBACK(_event_button_press), table);
+
+  table->gesture_button_primary = gtk_gesture_multi_press_new(GTK_WIDGET(table->widget));
+  gtk_gesture_single_set_button(GTK_GESTURE_SINGLE(table->gesture_button_primary),
+                                GDK_BUTTON_PRIMARY);
+  g_signal_connect(G_OBJECT(table->gesture_button_primary), "pressed",
+                   G_CALLBACK(_event_button_press_primary), table);
+  g_signal_connect(G_OBJECT(table->gesture_button_primary), "released",
+                   G_CALLBACK(_event_button_release_primary), table);
+
```

Comment:
It seems likely that this sequence of four calls will be repeated over and over for each  button-press handler.  That suggests packaging them into a function called something like `dtgtk_button_handler_new` which would take as arguments the GtkWidget, the specification of which button, the press and release callbacks, and the user-data pointer.  With that abstraction in place, the above code would read
```
table->gesture_button_primary = dtgtk_button_handler_new(table->widget, GDK_BUTTON_PRIMARY, _event_button_press_primary, _event_button_release_primary, table);
```

Reply:
Such a message might be of general use. Do you (or anyone else) has a short hint where to place such a method? I don't have a general overview of the whole code structure for now.

Reply:
`src/dtgtk/button.h` seems the obvious place.